test_runner/regress/test_wal_acceptor.py

```diff
@@ -526,25 +518,21 @@ def test_s3_wal_replay(neon_env_builder: NeonEnvBuilder, remote_storage_kind: Re
 
             last_lsn = query_scalar(cur, "SELECT pg_current_wal_flush_lsn()")
 
-    pageserver_lsn = env.pageserver.http_client().timeline_detail(
-        uuid.UUID(tenant_id), uuid.UUID((timeline_id))
-    )["local"]["last_record_lsn"]
+    ps_cli = env.pageserver.http_client()
+    pageserver_lsn = ps_cli.timeline_detail(uuid.UUID(tenant_id), uuid.UUID((timeline_id)))[
+        "local"
+    ]["last_record_lsn"]
     lag = lsn_from_hex(last_lsn) - lsn_from_hex(pageserver_lsn)
     log.info(
         f"Pageserver last_record_lsn={pageserver_lsn}; flush_lsn={last_lsn}; lag before replay is {lag / 1024}kb"
     )
 
-    # replace pageserver with a fresh copy
     pg.stop_and_destroy()
-    env.pageserver.stop()
 
-    log.info(f"Removing current pageserver state at {pageserver_tenants_dir}")
-    shutil.rmtree(pageserver_tenants_dir)
-    log.info(f"Copying fresh pageserver state from {pageserver_fresh_copy}")
-    shutil.move(pageserver_fresh_copy, pageserver_tenants_dir)
+    # recreate timeline on pageserver from scratch
+    ps_cli.timeline_delete(UUID(tenant_id), UUID(timeline_id))
+    ps_cli.timeline_create(UUID(tenant_id), UUID(timeline_id))
```

Comment:
It also may be useful to "simulate" a real scenario when pageserver has crashed and fully replaced with a fresh one. To do that we can restart/delete pageserver dir between `timeline_delete` and `timeline_create`. But the current test is ok already, so this is just an idea for additional test.